src/Adapter/Product/Combination/QueryHandler/GetEditableCombinationsListHandler.php

```
@@ -108,8 +109,11 @@ public function __construct(
      */
     public function handle(GetEditableCombinationsList $query): CombinationListForEditing
     {
+        $shopId = $query->getShopId();
```

Actually similarly to what was done in `GetProductForEditing` the query should rely on a `ShopConstraint` not just a shopId this will already include more potential use cases for the future if someday we try to improve the All shops view.

It's not that complex the idea is:
- if shop constraint is single shop do like you did here
- if shop constraint is allShop we need to get the product's default shop, the rest is the same
- if shop constraint is a group one we throw an exception

It has been handled in this method:
https://github.com/PrestaShop/PrestaShop/blob/30b0e97cd676650ab49bfd11e3d0b486a9ed9040/src/Adapter/Product/Repository/ProductMultiShopRepository.php#L172

Which should probably be "copied" into a more generic `getShopIdByShopConstraint` more fitted to your use case.

---

Oh but actually you already implemented `CombinationMultishopRepository::getByShopConstraint` so all good

---

Can you please change the command to handle a ShopConstraint instead of a single shop ID. You don't even need to handle the allShops case you can throw an exception with a `Not handled yet` but at least the command constructor is already ok and we won't need to change it and have a breaking change later.